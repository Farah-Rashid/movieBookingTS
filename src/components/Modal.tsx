import React from 'react'
import close from "../Assets/Xclose.png"
import {
  ModalStyle,
  CloseButton,
  DataFlex,
  MoviePic,
  ModalH1,
  ModalH3,
  ModalH5,
  ModalSmall,
  ModalDiv,
  ModalDivSeat,
  ModalSeatH1,
} from "../styles/Booking.styled";

type Props = {
movie:string;
selectedSeats:string[];
setSelectedSeats:(arr: string[]) => void;
setModalVisible:(arr: boolean) => void;
setBgClick:(arr:boolean)=>void;
}

const Modal = ({movie,selectedSeats,setModalVisible,setSelectedSeats,setBgClick}: Props) => {
  const modalClose = () => {
   setModalVisible(false);
   setSelectedSeats([]);
   setBgClick(true);
    };

  return (
    <ModalStyle>
    <CloseButton onClick={modalClose} src={close} alt="close" />
    <ModalH1>Thank you for booking</ModalH1>
    <ModalH3>Order Summary</ModalH3>
    < ModalDiv>

      <MoviePic src={movie} alt="movie" />

      <article>
        <h3>Seats:</h3>
        <ModalDivSeat>
          {selectedSeats.map((seat, index) => {
            return <ModalSeatH1 key={index}>{seat},</ModalSeatH1>;
          })}
          
        </ModalDivSeat>
        <DataFlex>
          <article>
            <ModalH5>{selectedSeats.length}*250</ModalH5>
            <br />
            <ModalSmall>CGST(12%)</ModalSmall>
            <br />
            <ModalSmall>SGST(12%)</ModalSmall>
          </article>
          <article style={{textAlign:"right"}}>
            <ModalH5>{selectedSeats.length*250}</ModalH5>
            <br />
            <ModalSmall>{12*(1/100)*(selectedSeats.length*250)}</ModalSmall>
            <br />
            <ModalSmall>{12*(1/100)*(selectedSeats.length*250)}</ModalSmall>
          </article>
        </DataFlex>
        <hr />
        <DataFlex>
          <ModalH5>Total</ModalH5>
          <ModalH5>{(selectedSeats.length*250)+(12*(1/100)*(selectedSeats.length*250))*2}</ModalH5>
        </DataFlex>
      </article>
    </ModalDiv>
  </ModalStyle>
  )
}

export default Modal;