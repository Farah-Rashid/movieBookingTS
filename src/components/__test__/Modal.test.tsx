import Modal from '../Modal';
import { render, screen, fireEvent } from '@testing-library/react';


const data = ['A1', 'B1', 'C1', 'D1', 'A2', 'B2', 'C3']
const mockSetVisible = jest.fn();
const mockSeatSelector = jest.fn()
describe("Modal test", () => {
  test('should take a snapshot', () => {
    const { container } = render(<Modal setSelectedSeats={mockSeatSelector} setModalVisible={mockSetVisible} selectedSeats={data} />)
    expect(container).toMatchSnapshot()
  })
  test("testing modal closing button", () => {
    render(<Modal setSelectedSeats={mockSeatSelector} setModalVisible={mockSetVisible} selectedSeats={data} />)
    const closeButton = screen.getByTestId<HTMLImageElement>("closeButton")
    fireEvent.click(closeButton);
    expect(mockSetVisible).toHaveBeenCalled()
  });
})