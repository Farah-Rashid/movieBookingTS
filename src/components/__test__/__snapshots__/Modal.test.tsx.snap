// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test should take a snapshot 1`] = `
<div>
  <article
    class="sc-ksZaOG bJBVYL"
  >
    <article
      class="sc-jSMfEi bmWsQR"
    >
      <img
        alt="close"
        class="sc-gKXOVf cYuPVG"
        data-testid="closeButton"
        src="Xclose.png"
      />
      <h1
        class="sc-ftvSup habaGD"
      >
        Thank you for booking
      </h1>
      <h3
        class="sc-papXJ htdLFC"
      >
        Order Summary
      </h3>
      <article
        class="sc-crXcEl fEfpzX"
      >
        <img
          alt="movie"
          class="sc-iBkjds liIVTu"
          src="https://image.tmdb.org/t/p/w500/null"
        />
        <article>
          <h3
            class="sc-hAZoDl hqlHun"
          >
            Seats:
          </h3>
          <article
            class="sc-evZas dNqGqz"
          >
            <h1
              class="sc-breuTD hAhIDx"
            >
              A1,B1,C1,D1,A2,B2,C3
            </h1>
          </article>
          <article
            class="sc-iqcoie dLFwIy"
          >
            <article>
              <h5
                class="sc-jqUVSM cmsqhK"
              >
                7
                *250
              </h5>
              <br />
              <small
                class="sc-kDDrLX dtSAIy"
              >
                CGST(12%)
              </small>
              <br />
              <small
                class="sc-kDDrLX dtSAIy"
              >
                SGST(12%)
              </small>
            </article>
            <article>
              <h5
                class="sc-jqUVSM cmsqhK"
              >
                1750
              </h5>
              <br />
              <small
                class="sc-kDDrLX dtSAIy"
              >
                210
              </small>
              <br />
              <small
                class="sc-kDDrLX dtSAIy"
              >
                210
              </small>
            </article>
          </article>
          <hr />
          <article
            class="sc-iqcoie dLFwIy"
          >
            <h5
              class="sc-jqUVSM cmsqhK"
            >
              Total
            </h5>
            <h5
              class="sc-jqUVSM cmsqhK"
            >
              2170
            </h5>
          </article>
        </article>
      </article>
    </article>
  </article>
</div>
`;
